src/Raven.Server/Documents/ETL/EtlTransformer.cs

```diff
@@ -140,22 +140,31 @@ private JsValue LoadAttachment(JsValue self, JsValue[] args)
 
             var attachmentName = args[0].AsString();
             var loadAttachmentReference = CreateLoadAttachmentReference(attachmentName);
-
+            Attachment attachment = null;
             if ((Current.Document.Flags & DocumentFlags.HasAttachments) == DocumentFlags.HasAttachments)
             {
-                var attachment = Database.DocumentsStorage.AttachmentsStorage.GetAttachment(Context, Current.DocumentId, attachmentName, AttachmentType.Document, null);
+                attachment = Database.DocumentsStorage.AttachmentsStorage.GetAttachment(Context, Current.DocumentId, attachmentName, AttachmentType.Document, null);
 
                 if (attachment == null)
-                    return JsValue.Null;
-
-                AddLoadedAttachment(loadAttachmentReference, attachmentName, attachment);
+                    UpdateEmptyAttachmentInfo();
             }
             else
             {
-                return JsValue.Null;
+                UpdateEmptyAttachmentInfo();
             }
 
+            AddLoadedAttachment(loadAttachmentReference, attachmentName, attachment);
+
             return loadAttachmentReference;
+
+            void UpdateEmptyAttachmentInfo()
+            {
+                loadAttachmentReference = (JsNull)Activator.CreateInstance(typeof(JsNull), true);
```

Comment:
not sure I understand this? why not `JsValue.Null` ?

Reply:
The problem here is that we need to be able to support:

```
var attachment = loadAttach('missing');
if(!attachment) { ... }
```

If you return another value, this will not be a null value in the JS script.